### ISSUE TITLE
Structuring campaign themes and campaigns

### ISSUE THREAD

**Issue by aschrijver** [humanetech-community/humanetech-community-awareness#6]
This is a follow-up to [Reality Shock Project Structure](https://github.com/humanetech-community/humanetech-community-awareness/issues/5), with the purpose to generalize the campaign management structure.

We both need a clear project structure and 'information model' (terminology / relationships) on how to set up campaigns.

Requirements:
- Consistency across campaigns
- Improving readability, get quick insights / overview
- Clear workflow for cooperating
- Structure allows for easy division of tasks
- Templates (for easy submissions)
- Documentation (help and clarification

This is a brainstorming issue, after which we'll document the outcomes, and improve from there. So let's investigate a possible way of structuring..

## Information model

There are 5 different information types, that will be reflected in the repository by separate documents:

- **Theme**
  - **Campaign**
    - **Deliverable**
    - **Strategy**
    - **Funding**

Not all of them are required to exist:

- Strategy and Funding documents are only created when the complexity of the campaign require it.

### Theme

Themes are the overarching categories for one or more campaigns

| Field | Description |
| :---: | :--- |
| `Name` | Short name that reflects the gist of the theme, used as working title |
| `Summary` | Clear and short explanation in 1-3 lines of text |
| `Topics` | Bullet list with (humane) tech topics and/or harms that are targeted |
| `Objectives` | Bullet list of what we want to achieve with the campaigns in this theme |
| `Description` | A longer, more elaborate description (couple of paragraphs) |
| `Strategy` | (optional) If there is a common strategy, outline it here |
| `Campaigns` | Hyperlinked ToC to past, ongoing and future campaigns |
| `Resources` | (optional) Links to relevant subfolders, files and external information |

- Theme-related content is contained in a subfolder as: `/themes/[theme-name]`
- The fields form the _theme template_ of the root document that outlines the theme (`README.md`)
- Proposing a new theme involves creating an Issue with an _issue template_ with same fieds
- When theme details are sufficiently discussed they are moved to `README.md` in theme subfolder

### Campaign

Campaigns are the central information item. The goals of the campaign is to create output, in the form of one ore more Deliverables, that are brought to a target Audience using the campaign Strategy on order to reach the campaign Goals.

| Field | Description |
| :---: | :--- |
| `Info` | Name, Theme, Status, Affiliation, Original idea, Coordinator, Forum link |
| `Summary` | Clear and short explanation in 1-3 lines of text |
| `Goals` | Bullet list of the intended effects of the campaign |
| 'Audience` | The demographic audience the campaign is targeted to |
| `Success criteria` | (optional) Bullet list detailing how success is measured |
| `Retrospective` | (optional) Analysis of results after campaign has ended |
| `Description` | A longer, more elaborate description (couple of paragraphs) |
| `Deliverables` | Hyperlinked ToC with campaign deliverables + short summary per entry |
| `Strategy` | Outlining the strategy to follow to attain the success criteria |
| `Funding` | (optional) Financial requirements, required budget, ways to obtain funds |
| `Milestones` | (optional) Summary of past and future milestones for the campaign |
| `Resources` | (optional) Links to relevant subfolders, files and external information |

- Similar to themes, a Campaign is documented in a 'README.md' that starts as an issue
- All campaign-related files are contained in a subfolder `/campaigns/[campaign-name]`
- The Status field can have following values: Ideation, Preparing, Launched, Finished
- The Affiliation field can be either: Official or Unofficial

If the campaign is complex, e.g. needs intricate planning, multiple deliverables, significant funding, then Stratey and Funding are detailed further in separate documents and hyperlinked from the campaign.

### Deliverable

Deliverables are the result of a campaign. They are produced during campaign planning, and are then brought to the target audience.

| Field | Description |
| :---: | :--- |
| `Info` | Name, Campaign, Status, Affiliation, Original idea, Coordinator, Forum link |
| `Type` | E.g. Video, Website, Image, Blog, Press Release, Meetup, Advert, etc. |
| `Summary` | Clear and short explanation in a single line (rest goes to Concept field) |
| `Goals` | Bullet list of the intended purpose of the deliverable |

Additional fields depend on the deliverable type. Let's mention only video for now.

#### Video Deliverable

| Field | Description |
| :---: | :--- |
| `Concept` | Describe the concept, the moral, underlying idea of the video |
| `Strategy` | (optional) Additional strategy requirements to that outlined in campaign |
| `Script` | Summary of the script of the video (couple of paragraphs) |
| `Duration` | Indicator on how long the video should be |
| `Music` | Plan for the music that is likely to be used in the video |
| `Production` | Summary of high-level plan on how to produce the video |
| `Funding` | Funding requirements, budget needed to produce the video |
| `Team` | List of all people involved and their role/tasks (the eventual 'credit roll') |
| `Resources` | (optional) Links to relevant subfolders, files and external information |

- Deliverables go to a subfolder of the campaign they belong to: `/campaign/[campaign-name]/[deliverable-name]`
- It is likely that there will be multiple documents with additional details on production, music, funding, etc.
- The Status field can have following values: Ideation, Preparing, Launched, Finished
- The Affiliation field can be either: Official or Unofficial

### Strategy, Funding

Let's wait with creating information models and templates for these, until we know more about the work method and process, etc.

## Project structure

At the highest level the project structure (the files section) is as follows:

- `themes` - All themes
- `campaigns` - All campaigns
- `stragegy` - Common strategy resources, guidelines, documentation
  - `marketing` - Multi-channel & social media marketing, promotion, distribution
  - `crowdfunding` - General info on funding campaigns
  - `deliverables` - Information on creating various deliverable types
    - `video` - How-to's, resources for producing video deliverables 
- `resources` - General resources that can be of interest for campaigns and deliverables

**Comment by georgeshk** [comment]
My goodness! That is very impressive work. Seems all good to me. To make sure I understand correctly, if we have a campaign that includes several videos, we would need to refer it to one of the themes and store it under /Campaigns. Under this campaign, we would then have a /Deliverables folder, under which would come each separate video. In say /Video-1 folder, we would have a readme file that shows all the required fields (note that the field "concept" should be defined not as a working name - already given in the field "name", but as the video's "concept" as defined by borja (one short paragraph).

**Comment by aschrijver** [comment]
> I understand correctly

Yes, that is the gist. We should not make the folder hierarchy too deep (unwieldy), but maybe an extra level for deliverable type is need, otherwise the subfolder name should indicate the type. Options:

1. `campaigns/phone-zombies/video-app-infection`
2. `campaigns/phone-zombies/video/app-infection`

My preference swaps between 1. and 2.

---

I've added 2 fields to the information model above: `Campaign > Retrospective` and `Video > Duration`.

---

I have changed the labeling system to make it clearer, simpler. I'll adjust issue tags and titles.

And I will create issue templates corresponding to the information model (note: you don't have to use table format, but headers instead, easier; it was just used here for the information model).

**Comment by aschrijver** [comment]
I have created the _Issue Template_ for themes. You can test it creating a new issue and selecting the 'Theme proposol' issue type (but do not save the issue, unless making a real proposal).

**Comment by georgeshk** [comment]
Propose to call deliverables folders as following example: "video(app-infection)", i.e. "[type]([working-title])", or anything that specifies both the type and working title.

As to structuring in general, up to a few admins to make sure all is consistent when adding final products of discussions to the folders. Let us make it simple for the majority of contributors.

**Comment by aschrijver** [comment]
I prefer option 2 then. `(` and `)` are reserved characters in URL's and also do not work well across OS'es.

Yes, few admins, otherwise it will be a mess. In Github terms they are called 'Maintainers' btw.

I created 2 issue templates, with 2 more to come tomorrow. Did not update issue content accordingly (but that was just nice as example and for 'clean slate', not _really_ needed).

**Comment by aschrijver** [comment]
TODO: Add Status fields, and update existing documents.
